src/Language/PureScript/CodeGen/JS.hs

```diff
@@ -234,25 +232,24 @@ moduleToJs (Module _ coms mn _ imps exps reExps foreigns decls) foreign_ =
     return $ AST.Function Nothing Nothing jsArg (AST.Block Nothing [AST.Return Nothing ret])
   valueToJs' e@App{} = do
     let (f, args) = unApp e []
-    args' <- mapM valueToJs args
-    let _ = unsafePerformIO (print (unApp (head args) (tail args)) >> pure f)
     case f of
       Var _ (Qualified _ (Ident "apply")) -> do
-        let (g, args'') = unApp (head args) (tail args)
-        args''' <- mapM valueToJs args''
-        case g of
-          Var (_, _, _, Just IsNewtype) _ -> return (head args''')
-          Var (_, _, _, Just (IsConstructor _ fields)) name | length args'' == length fields ->
-            return $ AST.Unary Nothing AST.New $ AST.App Nothing (qualifiedToJS id name) args'''
-          _ -> flip (foldl (\fn a -> AST.App Nothing fn [a])) args''' <$> valueToJs f
-      Var (_, _, _, Just IsNewtype) _ -> return (head args')
-      Var (_, _, _, Just (IsConstructor _ fields)) name | length args == length fields ->
-        return $ AST.Unary Nothing AST.New $ AST.App Nothing (qualifiedToJS id name) args'
-      _ -> flip (foldl (\fn a -> AST.App Nothing fn [a])) args' <$> valueToJs f
+        let (f', args') = unApp (head args) (tail args)
```

Comment:
I'm not very confident with using `head args` here, since it assumes that `args` always has an element because of `"apply"` – but I don't think this is necessarily a problem, since the `IsNewtype` case assumes it (or at least `args'`) to have an element either way.